### PR TITLE
Clean-up Python #4815 - P1.13: Chem\Pharm2D & Chem\Pharm3D

### PR DESCRIPTION
**Pull request by IchiruTake**
This is the sub-PR breaking from the large PR 4815: https://github.com/rdkit/rdkit/pull/4815

------
Updated Changes:
- [Remark]: Optimize `DownsampleBoundsMatrix()` in `EmbedLib.py`. The execution is varied depended on the input rather than consuming the worst-case scenario
- [Bug]: `DownsampleBoundsMatrix()` in `EmbedLib.py` now returns the same datatype as input rather than always casting to `np.float64`
- [New]: Drop/Remove one doctest in `OrderTriangle()` as the result of duplication
- [New]: Function `CountUpTo()` use `NumCombinations()` one time if `verbose`  instead of two in the previous examples
- [New]: Remove function `_fact()` of user in `Pharm2D\Utils.py`. Now we used `math.comb()` and `math.factorial()` for computation if python version is 3.8+. Otherwise, use user-defined function to speed up. The current algorithm shared high similarity to the `math.comb()` and is able to prevent numeric overflow. Extra documentation is included.
- [New]: Function `GetUniqueCombinations()` is now faster. However, the acceptance is likely to be reverted if required

### CODE REVIEW

File: rdkit/Chem/Pharm3D/EmbedLib.py [hunk 1]
```diff
@@ -178,9 +170,9 @@ def EmbedMol(mol, bm, atomMatch=None, weight=2.0, randomSeed=-1, excludedVolumes
   nAts = mol.GetNumAtoms()
   weights = []
   if atomMatch:
-    for i in range(len(atomMatch)):
-      for j in range(i + 1, len(atomMatch)):
-        weights.append((i, j, weight))
+    atomMatchSize: int = len(atomMatch)
```
Comment (greglandrum) [inline rdkit/Chem/Pharm3D/EmbedLib.py:173]: we don't use type hints in RDKit code.

**Review by greglandrum — CHANGES_REQUESTED** [review]
Please remove the type hint

File: rdkit/Chem/Pharm3D/EmbedLib.py (hunk 1 above)
Comment (IchiruTake) [reply rdkit/Chem/Pharm3D/EmbedLib.py]: Yes, I have fixed it.

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
Thanks @IchiruTake : I will merge this after the CI builds finish.

**Comment by IchiruTake** [comment]
> Thanks @IchiruTake : I will merge this after the CI builds finish.

Thank you for your support. Can you help me reviewed PR #4870?

**Pull request merged**